internal/bminventory/inventory.go

```diff
@@ -1694,6 +1694,18 @@ func checkPlatformWrongParamsInput(platform *models.Platform, userManagedNetwork
 	return nil
 }
 
+func doesPlatformDifferentThanBaremetalOrNone(platform *models.Platform, cluster *common.Cluster) bool {
+	if platform != nil && *platform.Type != models.PlatformTypeBaremetal && *platform.Type != models.PlatformTypeNone {
+		return true
+	}
+
+	if platform == nil && *cluster.Platform.Type != models.PlatformTypeBaremetal && *cluster.Platform.Type != models.PlatformTypeNone {
```

Comment:
why do we need to check both cluster.Platform.Type & platform.Type?

Reply:
Because if the user didn't provide the platform in the request, and the platform is already set to let say vsphere, we should honer the request.
We only care about the UMN & platform alignment when the platform is Baremetal or None.

Reply:
Because if the cluster platform type is currently set to vsphere and we just want to update UMN, it's once case .
Another case is where we set them both together.
On both of those cases we do not want to go and check all the if cases for none/bm platform